packages/typescript-estree/src/parser.ts

```diff
@@ -334,42 +402,19 @@ function generateAST<T extends ParserOptions = ParserOptions>(
       throw convertError(error);
     }
   }
-
-  return {
-    estree,
-    program: shouldProvideParserServices ? program : undefined,
-    astMaps: shouldProvideParserServices
-      ? astMaps!
-      : { esTreeNodeToTSNodeMap: undefined, tsNodeToESTreeNodeMap: undefined }
-  };
-}
-
-//------------------------------------------------------------------------------
-// Public
-//------------------------------------------------------------------------------
-
-export const version: string = packageJSON.version;
-
-export function parse<T extends ParserOptions = ParserOptions>(
-  code: string,
-  options?: T
-) {
-  if (options && options.errorOnTypeScriptSyntacticAndSemanticIssues) {
-    throw new Error(
-      `"errorOnTypeScriptSyntacticAndSemanticIssues" is only supported for parseAndGenerateServices()`
-    );
-  }
-  return generateAST<T>(code, options).estree;
-}
-
-export function parseAndGenerateServices(code: string, options: ParserOptions) {
-  const result = generateAST(code, options, /*shouldGenerateServices*/ true);
+  /**
+   * Return the converted AST and additional parser services
+   */
   return {
-    ast: result.estree,
+    ast: estree,
     services: {
-      program: result.program,
-      esTreeNodeToTSNodeMap: result.astMaps.esTreeNodeToTSNodeMap,
-      tsNodeToESTreeNodeMap: result.astMaps.tsNodeToESTreeNodeMap
+      program: shouldProvideParserServices ? program : undefined,
+      esTreeNodeToTSNodeMap: shouldProvideParserServices
```

Comment:
instead of checking flag we should check `astMaps`, you will be able to remove null assertion `!`